addons/account/models/company.py

```diff
@@ -342,9 +342,10 @@ def _validate_fiscalyear_lock(self, values):
 
     def _get_user_fiscal_lock_date(self):
         """Get the fiscal lock date for this company depending on the user"""
-        lock_date = max(self.period_lock_date or date.min, self.fiscalyear_lock_date or date.min)
+        # We need to use sudo, since we might not have access to a parent company.
+        lock_date = max(self.sudo().period_lock_date or date.min, self.sudo().fiscalyear_lock_date or date.min)
         if self.user_has_groups('account.group_account_manager'):
-            lock_date = self.fiscalyear_lock_date or date.min
+            lock_date = self.sudo().fiscalyear_lock_date or date.min
         if self.parent_id:
             lock_date = max(lock_date, self.parent_id._get_user_fiscal_lock_date())
```

Comment:
Putting the `sudo` on `self.parent_id` makes the code more readable, and it means that at least we have access to `self`

Reply:
It looks cleaner indeed. It's less obvious that way though that we might be working with a sudo() record, but since we are calling the same method, I guess it is clear enough.